programs/manifest/tests/cases/global.rs

```diff
@@ -550,10 +555,151 @@ async fn global_insufficient() -> anyhow::Result<()> {
+        )
+        .await;
+        test_fixture
+            .usdc_mint_fixture
+            .mint_to(&token_account_fixture.key, 1_000_000)
+            .await;
+
+        // There are bugs with the first tx looking like it fails but actually
+        // succeeds. Just continue on.
+        let _ = send_tx_with_retry(
+            Rc::clone(&test_fixture.context),
+            &[
+                transfer(&payer, &new_keypair.pubkey(), 10_000_000),
+                global_add_trader_instruction(
+                    &test_fixture.global_fixture.key,
+                    &new_keypair.pubkey(),
+                ),
+                global_deposit_instruction(
+                    &test_fixture.global_fixture.mint_key,
+                    &new_keypair.pubkey(),
+                    &token_account_fixture.key,
+                    &spl_token::id(),
+                    1_000_000,
+                ),
+            ],
+            Some(&payer_keypair.pubkey()),
+            &[&payer_keypair, &new_keypair],
+        )
+        .await;
+    }
+
+    // Adds with payer
+    test_fixture.global_add_trader().await?;
+
+    let evictee_account_keypair: Keypair = Keypair::new();
+    let evictee_account_fixture: TokenAccountFixture = TokenAccountFixture::new_with_keypair(
+        Rc::clone(&test_fixture.context),
+        &test_fixture.global_fixture.mint_key,
+        &payer,
+        &evictee_account_keypair,
+    )
+    .await;
+
+    let evictor_account_keypair: Keypair = Keypair::new();
+    let evictor_account_fixture: TokenAccountFixture = TokenAccountFixture::new_with_keypair(
+        Rc::clone(&test_fixture.context),
+        &test_fixture.global_fixture.mint_key,
+        &test_fixture.second_keypair.pubkey(),
+        &evictor_account_keypair,
+    )
+    .await;
+    test_fixture
+        .usdc_mint_fixture
+        .mint_to(&evictor_account_fixture.key, 1_000_000)
+        .await;
+
+    // Second keypair is evicting first.
+    send_tx_with_retry(
+        Rc::clone(&test_fixture.context),
+        &[global_evict_instruction(
+            &test_fixture.global_fixture.mint_key,
+            &test_fixture.second_keypair.pubkey(),
+            &evictor_account_fixture.key,
+            &evictee_account_fixture.key,
+            &spl_token::id(),
+            1_000_000,
+        )],
+        Some(&test_fixture.second_keypair.pubkey()),
+        &[&test_fixture.second_keypair.insecure_clone()],
+    )
+    .await?;
+
+    test_fixture.global_fixture.reload().await;
+    let global_dynamic_account: DynamicAccount<GlobalFixed, Vec<u8>> =
+        test_fixture.global_fixture.global;
+
+    // First got emptied.
+    let balance_atoms: GlobalAtoms = global_dynamic_account.get_balance_atoms(&payer);
+    assert_eq!(balance_atoms, GlobalAtoms::ZERO);
+    let balance_atoms: GlobalAtoms =
+        global_dynamic_account.get_balance_atoms(&test_fixture.second_keypair.pubkey());
+    assert_eq!(balance_atoms, GlobalAtoms::new(1_000_000));
+
```

Comment:
are you going to test match after eviction?

Reply:
Can add a test for that. Lots of tests need to be more thorough so will go through all of them once program is finalized

Reply:
added test